packages/components/src/modifiers/hds-tooltip.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// Note: the majority of this code is a porting of the existing tooltip implementation in Cloud UI
+// (which was initially implemented in Structure)
+
+import Modifier from 'ember-modifier';
+import type { ArgsFor } from 'ember-modifier';
+
+import { assert } from '@ember/debug';
+import { registerDestructor } from '@ember/destroyable';
+
+import tippy, { followCursor } from 'tippy.js';
+import type {
+  HideAll as TippyHideAll,
+  Instance as TippyInstance,
+  Props as TippyProps,
+} from 'tippy.js';
+// used by custom SVG arrow:
+import 'tippy.js/dist/svg-arrow.css';
+
+export interface HdsTooltipModifierSignature {
+  Args: {
+    Positional: [string];
+    Named: {
+      options?: TippyProps;
+    };
+  };
+  Element: HTMLElement;
+}
+
+function cleanup(instance: HdsTooltipModifier): void {
+  const { interval, needsTabIndex, tooltip } = instance;
+  if (needsTabIndex) {
+    tooltip?.reference?.removeAttribute('tabindex');
+  }
+  clearInterval(interval);
+  tooltip?.destroy();
+}
+
+/**
+ *
+ * `Tooltip` implements a modifier that uses Tippy.js to display a tooltip.
+ *
+ * Sample usage:
+ * ```
+ * <div {{hds-tooltip 'Text' options=(hash )}}>Hover me!</div>
+ * ```
+ *
+ * @see https://atomiks.github.io/tippyjs
+ * @class TooltipModifier
+ *
+ */
+export default class HdsTooltipModifier extends Modifier<HdsTooltipModifierSignature> {
+  didSetup = false;
+  interval: number | undefined = undefined;
+  needsTabIndex = false;
+  tooltip: TippyInstance | undefined = undefined;
+
+  constructor(owner: unknown, args: ArgsFor<HdsTooltipModifierSignature>) {
+    super(owner, args);
+    registerDestructor(this, cleanup);
+  }
+
+  hideOnEsc = {
+    name: 'hideOnEsc',
+    defaultValue: true,
+    fn({ hide }: { hide: TippyHideAll }) {
+      function onKeyDown(event: KeyboardEvent): void {
+        if (event.keyCode === 27) {
```

Comment:
nit: I know this was not introduced in this PR, but since `keyCode` is deprecated we can move to use `key`
```suggestion
        if (event.key === 'Escape') {
```

Reply:
sounds good, will do!